x-pack/plugins/file_upload/public/util/indexing_service.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import _ from 'lodash';
+import { i18n } from '@kbn/i18n';
+import { getIndexPatternService, getHttp } from '../kibana_services';
+
+export const getExistingIndexNames = _.debounce(
+  async () => {
+    let indexes;
+    try {
+      indexes = await getHttp().fetch({
+        path: `/api/index_management/indices`,
+        method: 'GET',
+      });
+    } catch (e) {
+      // Log to console. Further diagnostics can be made in network request
+      // eslint-disable-next-line no-console
+      console.error(e);
+    }
+    return indexes ? indexes.map(({ name }: { name: string }) => name) : [];
+  },
+  10000,
```

Comment:
Where did 10000 debounce value come from? Guess it does not really matter since this will get replaced with call to /internal/file_upload/index_exists in the next few days.

Reply:
Where all debounce values come from, it's made up :wink: . A 10 second turnaround seemed reasonable for getting new results but no strong feelings really. Agreed that if it's getting replaced, it's neither here nor there.